First version of the application.

gorealilian opened the pull request:
Supports:
* uploading a file into the local container
* retrieving the file through clickable links

gorealilian committed:
First version of the application. Supports:
* uploading a file into the local container
* retrieving the file through clickable links

gorealilian committed:
First version of the application. Supports:
* uploading a file into the local container
* retrieving the file through clickable links

stain committed:
Enable Travis CI for Java

stain commented on pom.xml:
```diff
@@ -0,0 +1,61 @@
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>com.uom.ac.uk</groupId>
```
Use `org.researchobject.roshow` as Maven `<groupId>`.

stain commented on src/main/java/roshow/RoApplication.java:
```diff
@@ -0,0 +1,26 @@
+package roshow;
```
Use `org.researchobject.roshow` as Java package name.

stain commented on src/main/java/roshow/RoController.java:
```diff
@@ -0,0 +1,83 @@
+        return "uploadForm";
+    }
+
+    @GetMapping("/files/{filename:.+}")
```
OK for first go, but for later you will need to include some kind of UUID or similar in the path `@GetMapping`s, so that multiple people can upload different ROs at the same time.

Even without a database you can just use the same UUID as part of the path of where you save the ROs.

stain commented on src/main/java/roshow/storage/UnzipUtility.java:
```diff
@@ -0,0 +1,77 @@
+        if (!destDir.exists()) {
+            destDir.mkdir();
+        }
+        ZipInputStream zipIn = new ZipInputStream(new FileInputStream(zipFilePath.toString()));
```
Later you may want to use the built-in `Path` [provider for ZIP](https://docs.oracle.com/javase/8/docs/technotes/guides/io/fsp/zipfilesystemprovider.html) instead of the lower-level `ZipInputStream`.

This would then provide a smooth upgrade path to potentially use the [taverna-robundle](https://github.com/apache/incubator-taverna-language/tree/master/taverna-robundle) `Path` provided instead, which adds additional methods for manipulating the  Research Object Manifest.  (And you would not need to keep around the unpacked ZIP content)

stain commented on src/test/java/roshow/FileUploadIntegrationTests.java:
```diff
@@ -0,0 +1,72 @@
+
+	@Ignore
+	@Test
+	public void shouldUploadFile() throws Exception {
```
Yay unit tests! But.. they say `@Ignore`? Tricked me!

stain reviewed (approved):
Looks good to me as a start! It did compile & run on my machine, just like in the demo you showed today.

I'll let @tomide1 review the code with you in person on how it works, then he can merge when satisfied.

See below for detailed comments.

tomide1 committed:
Merge pull request #2 from ResearchObject/feature/lilian

Feature/lilian

gorealilian committed:
Changelog:

* modified package name and group ID as per comments
* created a new UUID generator that changes the filenames of the uploaded archives to UUID's and adds them to the newly implemented H2 in memeory database
* changed the zipping utility architecture from low-level to a higher level as per @stain 's suggestion

gorealilian committed:
Merge remote-tracking branch 'origin/feature/lilian' into feature/lilian

tomide1 committed:
Merge remote-tracking branch 'remotes/origin/feature/lilian' into feature/telias

gorealilian committed:
Finishing up the H2 in-memory settings:
* now has two columns UUID (primary key) and Date (creation dateAndTime)
* upon uploading a research object the application now automatically renames the zip to a randomly created UUID and unzips it, also saves these details to the in-mem DB
* refactored the class distribution in packages to better show their resposibilities

tomide1 committed:
Merge remote-tracking branch 'origin/feature/lilian' into feature/telias
update local feature/telias with update to DB/UUID

gorealilian committed:
New MVC method using @RequestParam and some other helper methods to navigate inside a folder and display the manifest.json file on screen

tomide1 committed:
Merge remote-tracking branch 'origin/feature/lilian' into feature/telias
links to RO manifest

tomide1 committed:
changelog:
implement JsonReader facility
parse manifest file, extract & display top-level information on front-end

gorealilian committed:
Start separating controllers.
